packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -371,16 +386,48 @@ export default class CodeMode extends Component<Signature> {
           return;
         }
         try {
-          await this.importedModule.loaded;
-          let module = this.importedModule?.module;
-          if (module) {
-            let cards = cardsOrFieldsFromModule(module);
-            let elements: ElementInFile[] = cards.map((card) => {
-              return {
-                cardType: getCardType(this, () => card),
-                card: card,
-              };
-            });
+          if (isReady(this.openFile.current) && this.importedModule?.module) {
+            let module = this.importedModule?.module;
+            let cardsOrFields = cardsOrFieldsFromModule(module);
+            let elements: Element[] = [];
+            if (
+              this.openFile.current.url.endsWith('.json') &&
+              isCardDocumentString(this.openFile.current.content)
+            ) {
+              let cardOrField = cardsOrFields[0];
```

Comment:
This is a correct assumption right? adoptsFrom only expects a single export?

Reply:
i'm confused by this question--a card instance is not a module, it's a JSON file. A module is file that contains executable code. so i'm not sure that `module` is in this case.

Reply:
I agree its confusing.

This scenario covers the case when you open a json file which is a card. 
The inherits definition (the unhighlighted one) is actually loaded from a module. So despite us looking at Author/1, the author module resource is loaded in some way. 

**I'll try a way to unconfuse the json case. I've got an idea**

After investigation, the 0th index is definitiely the wrong assumption.

Reply:
I have done a simplification here https://github.com/cardstack/boxel/pull/699/commits/47e351d6d48c8f3877b53e41273572ebdbee60ed

Removing all ifs within resources that really should be looking at modules only and not considering the json card instance case